content/modules/rest/pages/business-logic.adoc

```diff
@@ -1,20 +1,208 @@
 = Business Logic
 
-== Service Invocation
+When interacting with the REST API it is oftentimes needed to have an application-level layer of business logic in place that represents the Invocation point for the API. It can be used for orchestration, validation, or other tasks that should happen when the API Client interacts with the system. The xref:rest:entities-api.adoc[Entities API] does not allow placing additional orchestration business logic in place. The Client in this case directly interacts with the entity layer.
+
+For exposing the ability to invoke business logic to the API Client, Jmix provides two ways of doing that:
+
+1. Services API
+2. Custom Controllers
+
+The Services API allows exposing an arbitrary Spring bean as an HTTP endpoint. In this case, Jmix will take care of the HTTP interactions like providing HTTP response codes, etc.
+
+On the other side, there is the ability to use custom HTTP controllers to invoke business logic from the client-side. The main difference is that in this case, it is also possible to influence the HTTP interactions (like status codes, security, etc.).
+
+== Services API
+
+Let's look at the first approach to expose business logic to the API Client: the Services API.
+
+To use a Spring bean as part of the Jmix Services API, it needs to fulfill the following criteria:
+
+1. The Spring bean needs to implement an Interface (the Service Interface), that exposes the method signatures.
+2. The Spring bean needs to be registered in the `rest-services.xml` configuration file
+
+=== Service Exposure
+
+To fulfill the first requirement of implementing a Java interface, let's look at the following example:
+
+The Service Interface allows to expose the methods via the REST Services API:
+
+[source,java]
+.CalculationService.java
+----
+public interface CalculationService {
+
+    String NAME = "sample_CalculationService"; // <1>
```

Comment:
We actually don't use constants for bean names in the framework and Studio doesn't generate them in the apps.
So this part should be rewritten.

Reply:
done